### PR TITLE
ci: add more extras to lockfiles

### PR DESCRIPTION
**Pull request by cpcloud**
This PR updates the conda-lock file generation to include some of the newer extras.

### CODE REVIEW

File: .github/workflows/conda-lock.yml [hunk 1]
```diff
@@ -71,15 +88,21 @@ jobs:
             --file "${python_version_file}" \
             --platform win-64 \
             --filename-template "${template}" \
+            -e bigquery \
```
Comment (gforsyth) [inline .github/workflows/conda-lock.yml:91]: if you want to add `duckdb` here, it's available on conda-forge now, so this should work.

**Review by gforsyth — APPROVED** [review]
looks good -- one comment, not blocking

File: .github/workflows/conda-lock.yml (hunk 1 above)
Comment (cpcloud) [reply .github/workflows/conda-lock.yml:91]: Huzzah!

**Comment by cpcloud** [comment]
I'm going to run this locally to make sure it works, moving to draft.

File: .github/workflows/conda-lock.yml (hunk 1 above)
Comment (cpcloud) [reply .github/workflows/conda-lock.yml]: Done!

**Comment by cpcloud** [comment]
Hm, encountering some issues with shapely, moving this back to draft.

**Comment by cpcloud** [comment]
Ok, fixed the shapely issue, conda-lock appears not to do package name normalization and requires an exact match.

**Comment by cpcloud** [comment]
Definitely meant to not include the shapely bump, reverting that, then merging on green.

**Pull request merged**